imagersite/imager_images/tests.py

```diff
@@ -0,0 +1,292 @@
+
+    def test_upload_image_adds_new_album_instance(self):
+        """Test new album instance."""
+        self.assertEqual(Album.objects.count(), 6)
+
+    def test_album_assigned_to_bob(self):
+        """Test that fred is album builder."""
+        test_bob = User()
+        test_bob.username = 'test_bob'
+        test_bob.save()
+        test_album = Album(user=test_bob)
+        test_album.save()
+        self.assertTrue(test_album.user.username == "test_bob")
+
+    def test_albums_created_are_private_by_default(self):
+        """Test default published status."""
+        self.assertEqual(self.album_one.published, "PB")
+
+    def test_new_album_choose(self):
+        """Test assigned published status."""
+        self.album_one.published = 'PV'
+        self.assertEqual(self.album_one.published, "PV")
+
+    def test_delete_user_with_albums_albums_removed(self):
+        """Test delete user all objects."""
+        self.assertTrue(Album.objects.count() == 6)
+        self.user.delete()
+        self.assertTrue(Album.objects.count() == 0)
+
+
+class TemplatesTests(TestCase):
+    """."""
+
+    def setUp(self):
+        """."""
+        self.client = Client()
+        user = User(
+            username='fred',
+            email='fred@fred.com'
+        )
+        user.set_password('temporary')
+        user.save()
+        self.user = user
+        photos = [PhotoFactory.build() for i in range(20)]
+        for photo in photos:
+            photo.user = user
+            photo.save()
+        user.save()
+
+    def test_to_libray_page_user_authenticated(self):
+        """Test library page redirect."""
+        self.client.login(username='fred', password='temporary')
+        response = self.client.get("/images/library/")
+
+        self.assertTrue(response.status_code == 200)
+        self.assertTemplateUsed(response, 'imager_images/library.html')
+
+    def test_to_photo_gallery_user_authenticated(self):
+        """Test photo gallery page redirect."""
+        self.client.login(username='Fred', password='temporary')
+        idx = self.user.photobuild.first().id
+        response = self.client.get("/images/photos/{}/".format(idx))
+
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'imager_images/photo_indiv.html')
+
+    def test_to_album_gallery_user_authenticated(self):
+        """Test album gallery page redirect."""
+        self.client.login(username='fred', password='temporary')
+        response = self.client.get("/images/albums/")
+        self.assertTrue(response.status_code == 200)
+        self.assertTemplateUsed(response, 'imager_images/album_gallery.html')
+
+    def test_when_no_image(self):
+        """Placeholder appears with default image."""
+        response = self.client.get(reverse_lazy('home'))
+        html = BeautifulSoup(response.content, 'html.parser')
+        self.assertTrue(html.find('image', {'src': "/MEDIA/user_images/animate0.png"}))
+
```

Comment:
You are missing a test for the list of all photos view.

I would like to see some more tests than just 200 response. Try using the response content to test if certain things are on the page.